src/preamble.js

```diff
@@ -1346,7 +1346,11 @@ FHEAP = new Float64Array(TOTAL_MEMORY);
 #endif
 #endif
 #if USE_TYPED_ARRAYS == 2
+#if POINTER_MASKING
+var buffer = new ArrayBuffer(TOTAL_MEMORY + 0x01000000);
```

Comment:
This concerns me. It seems more surprising than if we did the opposite, i.e., reserve space at the top. I suppose it's more convenient this way, though (so that `TOTAL_MEMORY` checks work as currently written)?

What can't work, though, is memory growth (i.e. if `TOTAL_MEMORY` isn't big enough, we can't grow the heap). Please add an assertion in emcc that `ALLOW_MEMORY_GROWTH` is not on.